File: hauyne-services/hauyne-service-admin/src/main/java/com/luoyx/hauyne/admin/sys/query/AuthorityQuery.java
package com.luoyx.hauyne.admin.sys.query;

import com.luoyx.hauyne.mybatisplus.query.PageQuery;
import com.luoyx.hauyne.validation.constraint.EnumCheck;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;
import lombok.Setter;

/**
 * 权限查询条件Query类
 *
 * @author 罗英雄
 */
@Getter
@Setter
public class AuthorityQuery {

    /**
     * 权限类型
     */
    @Schema(description = "权限类型")
    private String authorityType;

    /**
     * 权限名称
     */
    @Schema(description = "权限名称")
    private String authorityName;

    /**
     * 排序字段
     */
    @Schema(description = "排序字段")
    private String sortField;

    /**
     * 排序方式（1=升序，-1=降序）
     */
    @Schema(description = "排序方式（1=升序，-1=降序）", example = "-1")
    @EnumCheck(message = "排序方式枚举值不合法", enumClazz = PageQuery.SortOrderEnum.class, getterMethod = "getValue")
    private String sortOrder;
}

File: hauyne-front-end/src/app/pages/admin/authority/AuthorityQuery.ts
export interface AuthorityQuery {

    /**
     * 权限类型【? 表示这个属性有可能不存在】
     */
    authorityType?: string;

    /**
     * 权限名称【? 表示这个属性有可能不存在】
     */
    authorityName?: string;

    /**
     * 排序字段
     */
    sortField: string | null;

    /**
     * 排序方式（1=升序，-1=降序）
     */
    sortOrder: string | null;
}
